esp/clients/wsdfuaccess/wsdfuaccess.cpp

```diff
@@ -0,0 +1,66 @@
+/*##############################################################################
+
+    HPCC SYSTEMS software Copyright (C) 2018 HPCC Systems®.
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+############################################################################## */
+
+//Jlib
+#include "jliball.hpp"
+#include "dautils.hpp"
+
+#include "ws_dfu.hpp"
+
+#include "wsdfuaccess.hpp"
+
+
+bool WsDfuAccess_getMetaInfo(StringBuffer &metaInfoResult, const char *jobId, const char *logicalName, const char *access, unsigned expirySecs, const char *user, const char *token)
+{
+    // JCSMORE via config
+    const char *espHost = "localhost";
+    unsigned espServicePort = 8010;
+    const char *protocol = "http";
+    const char *service = "WsDfu";
```

Comment:
@wangkx - do you have a good suggestion of where these type of setting typically come from for a client service like this?

It could look them up from the environment...

Reply:
The 'service' should be hard-coded. If a command line tool uses this library, the espHost/espServicePort/protocol should be set inside the command. If it is used by an HPCC component, they should be read from configuration. So, they should be a part of WsDfuAccess_getMetaInfo() parameters.

Reply:
Is the service required in the URI? It might not be necessary.